data_structures/sliding_window.cpp

```diff
@@ -0,0 +1,159 @@
+
+/*
+ * @brief Sliding window implementation
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂